HLTrigger/Timer/src/ProcessCallGraph.cc

```diff
@@ -0,0 +1,324 @@
+  auto colormap = boost::make_container_vertex_map(colors);
+
+  // depht-first visit all vertices starting from the given module
+  boost::default_dfs_visitor visitor;
+  boost::depth_first_visit(graph_, module, visitor, colormap);
+
+  // count the visited vertices (the `black' ones) in order to properly size the
+  // output vector; then fill the dependencies with the list of visited nodes
+  unsigned int size = 0;
+  for (unsigned int i = 0; i < colors.size(); ++i)
+    if (boost::black_color == colors[i])
+      ++size;
+  std::vector<unsigned int> dependencies(size);
+  unsigned j = 0;
+  for (unsigned int i = 0; i < colors.size(); ++i)
+    if (boost::black_color == colors[i])
+      dependencies[j++] = i;
+  assert(size == j);
+
+  return dependencies;
+}
+
+// find the dependencies of all modules in the given path
+//
+// return two vector:
+//   - the first lists all the dependencies for the whole path
+//   - the second lists the one-after-the-last dependency index into the first vector for each module
+std::pair<std::vector<unsigned int>, std::vector<unsigned int>>
+ProcessCallGraph::dependencies(std::vector<unsigned int> const & path)
+{
+  std::vector<unsigned int> colors(boost::num_vertices(graph_));
+  auto colormap = boost::make_container_vertex_map(colors);
+
+  // first, find and count all the path's modules' dependencies
+  boost::default_dfs_visitor visitor;
+  for (unsigned int module: path)
+    boost::depth_first_visit(graph_, module, visitor, colormap);
+
+  unsigned int size = 0;
+  for (unsigned int i = 0; i < colors.size(); ++i)
+    if (colors[i] == 0)
+      ++size;
+
+  // allocate the output vectors
+  std::vector<unsigned int> dependencies(size);
+  dependencies.resize(0);
+  std::vector<unsigned int> indices(path.size());
+  indices.resize(0);
+
+  // reset the color map
+  for (unsigned int i = 0; i < colors.size(); ++i)
+    colors[i] = 0;
+
+  // find again all the dependencies, and record those associated to each module
+  struct record_vertices : boost::default_dfs_visitor {
+    record_vertices(std::vector<unsigned int> & vertices) :
+      vertices_(vertices) { }
+
+    void discover_vertex(unsigned int vertex, graph_type const& graph) {
+      vertices_.push_back(vertex);
+    }
+
+    std::vector<unsigned int> & vertices_;
+  };
+  record_vertices recorder(dependencies);
+
+  for (unsigned int module: path) {
+    // skip modules that have already been added as dependencies
+    if (colors[module] != boost::black_color)
+      boost::depth_first_visit(graph_, module, recorder, colormap);
+    indices.push_back(dependencies.size());
+  }
+
+  return std::make_pair(dependencies, indices);
+}
+
+// register a (sub)process and assigns it a "process id"
+// if called with a duplicate process name, returns the original process id
+unsigned int ProcessCallGraph::registerProcess(edm::ProcessContext const & context)
+{
+  static unsigned int id_ = 0;
```

Comment:
I suggest not using an underscore in the name since those are normally used for member variables. I personally use `s_` for statics.